src/main/java/net/minecraftforge/coremod/api/ASMAPI.java

```diff
@@ -361,35 +387,42 @@ public static MethodInsnNode findFirstMethodCallBefore(MethodNode method, Method
      * in the method provided. Only the first node matching is targeted, all other matches are ignored.
      *
      * @param method The method where you want to find the node
-     * @param type   The type of the old method node.
-     * @param owner  The owner of the old method node.
-     * @param name   The name of the old method node. You may want to use {@link #mapMethod(String)} if this is a srg
-     *               name
-     * @param desc   The desc of the old method node.
+     * @param type   The type of the old method node
+     * @param owner  The owner of the old method node
+     * @param name   The name of the old method node (you may want to use {@link #mapMethod(String)} if this is a srg
+     *               name)
+     * @param desc   The desc of the old method node
      * @param list   The list that should be inserted
      * @param mode   How the given code should be inserted
-     * @return True if the node was found, false otherwise
+     * @return True if the node was found and the list was inserted, false otherwise
      */
     public static boolean insertInsnList(MethodNode method, MethodType type, String owner, String name, String desc, InsnList list, InsertMode mode) {
-        Iterator<AbstractInsnNode> nodeIterator = method.instructions.iterator();
-        int opcode = type.toOpcode();
-        while (nodeIterator.hasNext()) {
-            AbstractInsnNode next = nodeIterator.next();
-            if (next.getOpcode() == opcode) {
-                MethodInsnNode castedNode = (MethodInsnNode) next;
-                if (castedNode.owner.equals(owner) && castedNode.name.equals(name) && castedNode.desc.equals(desc)) {
-                    if (mode == InsertMode.INSERT_BEFORE)
-                        method.instructions.insertBefore(next, list);
-                    else
-                        method.instructions.insert(next, list);
-
-                    if (mode == InsertMode.REMOVE_ORIGINAL)
-                        nodeIterator.remove();
-                    return true;
-                }
-            }
-        }
-        return false;
+        var insn = findFirstMethodCall(method, type, owner, name, desc);
+        if (insn == null) return false;
+
+        return insertInsnList(method, insn, list, mode);
+    }
+
+    /**
+     * Inserts/replaces a list after/before the given instruction.
+     *
+     * @param method The method where you want to insert the list
+     * @param list   The list that should be inserted
+     * @param mode   How the given code should be inserted
+     * @return True if the list was inserted, false otherwise
+     */
+    public static boolean insertInsnList(MethodNode method, AbstractInsnNode insn, InsnList list, InsertMode mode) {
```

Comment:
This method returns `boolean` to have parity with the older `insertInsnList` method. However, I think that it might be better to crash the game if a modder tries to insert a list into the instructions using an instruction that is not included in the method. What do you think?

Reply:
Having it return a boolena means the CoreMod itself can determine to crash or not